### PR TITLE
Fix backup worker crashing master on interruption

### PR DESCRIPTION
**Pull request by bf8086**
When a backup is interrupted, there is a chance that the resume callback cannot be canceled and thus happens after the server is already resumed by the journal reload process. While it makes sense for backup to crash the master if resume fails to complete and master cannot accept new journal entries, it is completely fine for master to continue if backup tries to resume an already resumed journal. This change updates the journal to not throw if someone attempts to resume after the journal is already resumed.

### CODE REVIEW

**Comment by bf8086** [comment]
@gpang @LuQQiu PTAL.

File: core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java [hunk 1]
```diff
@@ -537,8 +537,12 @@ public synchronized void suspend(Runnable interruptCallback) throws IOException
   public synchronized void resume() throws IOException {
     LOG.info("Resuming raft state machine");
     mInterruptCallback = null;
-    mJournalApplier.resume();
-    LOG.info("Raft state machine resumed");
+    if (mJournalApplier.isSuspended()) {
+      mJournalApplier.resume();
```
Comment (LuQQiu) [inline core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:541]: @bf8086 Can you provide a timeline for the double resume case?

Comment (bf8086) [reply core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:541]: Basically the backup starts and suspends the journal, then the journal resumes, interrupts the backup and reloads from the snapshot, and then the backup attempts to resume the journal again and hit this issue.

Comment (LuQQiu) [reply core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:541]: Got it, thanks!

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks!

**Comment by LuQQiu** [comment]
alluxio-bot, test this please

**Comment by LuQQiu** [comment]
No need to rerun tests since the integration tests passed for twice and only failed for the third run

**Comment by bf8086** [comment]
alluxio-bot, merge this please

**Comment by bf8086** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please.

**Pull request merged**